lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -384,6 +384,259 @@ class DecomposeAtenReshapeOp : public OpRewritePattern<AtenReshapeOp> {
+        }
+      } else if (equation.substr(index, 1).find(",") != std::string::npos) {
+        inputTokens.emplace_back(std::move(inputToken));
+      } else if ((index < (equation.size() - 1)) &&
+                 (equation.substr(index, 2).find("->") != std::string::npos)) {
+        inputTokens.emplace_back(std::move(inputToken));
+        currentVariable = kIsResult;
+        index++;
+      } else {
+        return rewriter.notifyMatchFailure(op, "unexpected character encountered");
+      }
+      index++;
+    }
+
+    SmallVector<Value> torchTensors;
+    if (!getListConstructElements(op.getTensors(), torchTensors)) {
+      return rewriter.notifyMatchFailure(
+          op, "input should comes from a PrimListConstructOp");
+    }
+
+    int64_t totalRank = torchTensors.size();
+    for (int i = 0; i < totalRank; i++) {
+      BaseTensorType tensorType = torchTensors[i].getType().cast<BaseTensorType>();
+      ArrayRef<int64_t> inputShape = tensorType.getSizes();
+      int64_t inputRank = inputShape.size();
+      SmallVector<Value> inputValueShape;
+      for (int j = 0; j < inputRank; j++) {
+        Value SizeInt = rewriter.create<AtenSizeIntOp>(
+                                loc, torchTensors[i], 
+                                rewriter.create<Torch::ConstantIntOp>(
+                                  loc, rewriter.getI64IntegerAttr(j)));
+        inputValueShape.push_back(inputShape[j] == kUnknownSize 
+                              ? SizeInt
+                              : rewriter.create<Torch::ConstantIntOp>(
+                                    loc, rewriter.getI64IntegerAttr(
+                                            inputShape[j])));
+      }
+
+      inputShapes.push_back(inputValueShape);
+    }
+
+
+    auto collectOperandDims =
+        [resultTokens](
+            SmallVector<Value> operandShape, SmallVector<char> operandTokens,
+            SmallVector<char> others, SmallVectorImpl<int64_t> &contractingDims,
+            SmallVectorImpl<int64_t> &batchingDims,
+            SmallVector<char> &dotResultTokens,
+            SmallVector<Value> &dotResultShape) {
+          llvm::SmallDenseSet<char> othersSet(others.begin(), others.end());
+          llvm::SmallDenseSet<char> resultTokensSet(resultTokens.begin(),
+                                                    resultTokens.end());
+          for (const auto &en : llvm::enumerate(operandTokens)) {
+            bool isResultToken = resultTokensSet.contains(en.value());
+            bool isOtherToken = othersSet.contains(en.value());
+            if (!isResultToken && isOtherToken) {
+              contractingDims.push_back(en.index());
+            } else if (isOtherToken) {
+              batchingDims.push_back(en.index());
+            } else {
+              dotResultTokens.push_back(en.value());
+              dotResultShape.push_back(operandShape[en.index()]);
+            }
+          }
+        };
+
+    for (int i = 0; i < totalRank - 1; i++) {
```

Comment:
Can you add a comment explaining why only loop until `i < totalRank - 1` (I would've expected to loop until `i < totalRank`)